RxCocoa/Common/Reactive.swift

```diff
@@ -7,27 +7,34 @@
 //
 
 /**
-We can use `Reactive` protocol as customization point for constrained protocol extensions.
+ Use `Reactive` proxy as customization point for constrained protocol extensions.
 
-General pattern would be:
+ General pattern would be:
 
+ // 1. Extend Reactive protocol with constrain on Self
+ // Read as: Reactive Extension where Self is a SomeType
+ extension Reactive where Self: SomeType {
+ // 2. Put any specific reactive extension for SomeType here
+ }
 
-    // 1. Conform SomeType to Reactive protocol
-    extension SomeType: Reactive {}
+ With this approach we can have more specialized methods and properties using
+ `Self` and not just specialized on common base type.
 
-    // 2. Extend Reactive protocol with constrain on Self
-    // Read as: Reactive Extension where Self is a SomeType
-    extension Reactive where Self: SomeType {
-        // 3. Put any specific reactive extension for SomeType here
-    }
+ */
 
+public struct Reactive<Base: AnyObject> {
```

Comment:
We can indeed remove the `AnyObject` requirement. I don't now if it really makes sense on a value type but who know. :)

For the `source` terminology I'm not convinced. I've came across the codebase looking for `let source =` declarations and everywhere, `source` is an `Observable`. Here it won't be on `Observable` but the object itself. It may confuse people if we mix up this two concepts behind the same term don't you think ?